src/formio.js

```diff
@@ -90,7 +91,14 @@ app.config([
 
 app.run([
   '$templateCache',
-  function($templateCache) {
+  '$rootScope',
+  'ngDialog',
+  function($templateCache, $rootScope, ngDialog) {
+    // Close all open dialogs on state change.
+    $rootScope.$on('$stateChangeStart', function() {
```

Comment:
Hi Travis,
Moved this to the resource component's scope.  It does get called when the state changes but I'm still not sure it's necessary since I never see the state change when the add resource dialog is open.